gdb/riscv-tdep.c

```diff
@@ -646,14 +640,55 @@ riscv_register_reggroup_p (struct gdbarch  *gdbarch,
     return regnum < RISCV_FIRST_FP_REGNUM;
   else if (reggroup == restore_reggroup || reggroup == save_reggroup)
     return regnum <= RISCV_LAST_FP_REGNUM;
-  else if (reggroup == system_reggroup)
-    return regnum >= RISCV_FIRST_CSR_REGNUM && regnum <= RISCV_LAST_CSR_REGNUM;
-  else if (reggroup == vector_reggroup)
+  else if (reggroup == system_reggroup) {
+    /* Only include CSRs that have aliases. */
+    if (regnum < RISCV_FIRST_CSR_REGNUM || regnum > RISCV_LAST_CSR_REGNUM)
+      return 0;
+    for (i = 0; i < ARRAY_SIZE (riscv_register_aliases); ++i) {
+	if (regnum == riscv_register_aliases[i].regnum)
+          return 1;
+    }
+    return 0;
+  } else if (reggroup == vector_reggroup)
     return 0;
   else
     internal_error (__FILE__, __LINE__, _("unhandled reggroup"));
 }
 
+static void
```

Comment:
did you really need to relocate this function ?  would be better to keep it where it was.